File: TestFile5/TestFile5.java
class TestFile5 {
	public static void main(String[] args){
		sendMessage(message -> System.out.println("My new fancy message"));
		sendMessage(message -> System.out.println("My second fancy message"));

		sendMessage(System.out::println);
		sendMessage(System.out::println);
	}


	interface PostMan<String> {
		void letterTime(String message);
	}
	private static void sendMessage(PostMan<String> message) {
		message.letterTime("Sending a default message");
	}
}

File: TestFile6/TestFile5.kt
fun main(vararg args: String){
	println(PostMan.letterTime("Fyre Festival!"))
}

class PostMan(val message: String){
	fun sendMessage(trap: String) = "$message... $trap"

	companion object{
		fun letterTime(trap: String) = PostMan("You've been invited to").sendMessage(trap)
	}
}